README.md

```diff
@@ -1,27 +1,508 @@
+        this.setState({ isModalOpen: true })
+    }
+    render () {
+        return (
+            <div>
+                <button onClick={this.openModal}>Open the modal!</button>
+                <Modal isOpen={this.state.isModalOpen} />
+            </div>
+        )
+    }
+}
+```
+
+Now if we click the `Open the modal!` button, we update the `<App />`
+component's internal state, thus enforcing an update on the `<App />` component.
+
+Since `this.state.isModalOpen` evaluates to `true` now, the modal is being
+opened:
+
+```js
+    render () {
+        return (
+            <div>
+                <button onClick={this.openModal}>Open the modal!</button>
+                <Modal isOpen={true} />
+            </div>
+        )
+    }
+```
+
+But what does our modal component actually look like? Up until now, we didn't
+talk about the `<Modal />` component's render method.
+
+As we just saw, the modal accepts an `isOpen` prop, which it then uses in order
+to add an `modal--is-open` class to its container:
+
+```js
+const Modal = ({ isOpen }) =>
+    <div className={isOpen ? 'modal modal--is-open' ? 'modal'}>
+        <button>close</button>
+
+        <p>Hello! I am a modal.</p>
+    </div>
+```
+
+Perfect! Now we can simply specify whether or not the modal is open using its
+`isOpen` prop.
+
+As you might have noticed, the modal also has a "close" button. Currently it
+doesn't do anything, so let's think about what should happen if the close button
+is being pressed:
+
+1. The user clicks the close button.
+2. The `<Modal />` component needs to update the `<App />` component's
+    `isModalOpen` state.
+
+For convenience, we're now going to add a `closeModal` method on the `<App />`
+components:
+
+```js
+class App extends React.Component {
+    // ...
+    openModal () {
+        this.setState({ isModalOpen: true })
+    }
+    closeModal () {
+        this.setState({ isModalOpen: false })
+    }
+    // ...
+}
+```
+
+The `<Modal />` component now needs to be able to call the `<App />` component's
+`closeModal` method. But React components can't directly access their ancestors,
+so instead we have to pass the `App`'s `closeModal` method as a handler to the
+`Modal`:
+
+```js
+const Modal = ({ isOpen, onClose }) =>
+    <div className={isOpen ? 'modal modal--is-open' ? 'modal'}>
+        <button onClick={onClose}>close</button>
+
+        <p>Hello! I am a modal.</p>
+    </div>
+```
+
+```js
+class App extends React.Component {
+    // ...
+    render () {
+        return (
+            <div>
+                <button onClick={this.openModal}>Open the modal!</button>
+                <Modal isOpen={this.state.isModalOpen} onClose={this.closeModal}/>
+            </div>
+        )
+    }
+    // ...
+}
+```
+
+And that's it! Now the modal is able to communicate with the `<App />` by
+calling the passed in handler.
+
+There is only one problem: When we click `close`, we get the following error:
+
+```
+TypeError: onClose is not a function
+```
```

Comment:
This is great — shows the students the pain of binding and the issues they'll run into. They should have already learned about this stuff in an earlier unit, but I would definitely keep it in (maybe make this section a little shorter to serve as a reminder of sorts)